### PR TITLE
State comparison - SSV runner/duties/newduty

### PR DESCRIPTION
**Pull request by olegshmuelov**
### Description
 - Introduce state comparison for sanity SSV runner/duties/newduty tests
 - Include explicit fields for FullHappyFlow test

reference to sanity tests https://github.com/bloxapp/ssv-spec/pull/162

### CODE REVIEW

File: ssv/spectest/tests/runner/duties/newduty/decided.go [hunk 1]
```diff
@@ -36,34 +36,38 @@ func PostDecided() tests.SpecTest {
...
+				PostDutyRunnerState:     postDecidedSyncCommitteeContributionSC().ExpectedState,
 				OutputMessages: []*types.SignedPartialSignatureMessage{
 					testingutils.PreConsensusContributionProofNextEpochMsg(ks.Shares[1], ks.Shares[1], 1, 1), // broadcasts when starting a new duty
 				},
 			},
 			{
 				Name:                    "sync committee",
-				Runner:                  finishRunner(testingutils.SyncCommitteeRunner(ks), &testingutils.TestingSyncCommitteeDuty),
+				Runner:                  decidedRunner(testingutils.SyncCommitteeRunner(ks), &testingutils.TestingSyncCommitteeDuty),
 				Duty:                    &testingutils.TestingSyncCommitteeDuty,
```
Comment (GalRogozinski) [inline ssv/spectest/tests/runner/duties/newduty/decided.go:50]: We are not using `testingutils.TestingSyncCommitteeDuty*NextEpoch*` here like the rest of the tests.
From my understanding of the tests, a good happy flow should look like the following:
```
	Runner:                  decidedRunner(testingutils.SyncCommitteeRunner(ks), &testingutils.TestingSyncCommitteeDuty),
	Duty:                    &testingutils.TestingSyncCommitteeDuty*NextEpoch*
```
This should startup with a decided runner for some epoch, and hten start a duty for the next.
But none of the tests follow this pattern.

On the other hand maybe we want to show that it doesn't matter for the spec if we follow the happy flow.
Then we should just have all tests be consistent..

wdyt @alonmuroch?

File: ssv/spectest/tests/runner/duties/newduty/not_decided.go [hunk 2]
```diff
@@ -48,35 +49,28 @@ func NotDecided() tests.SpecTest {
 				Name:                    "sync committee",
 				Runner:                  startRunner(testingutils.SyncCommitteeRunner(ks), &testingutils.TestingSyncCommitteeDuty),
 				Duty:                    &testingutils.TestingSyncCommitteeDuty,
-				PostDutyRunnerStateRoot: "c7602d60cec330c91db4923ddb71e3c74abd0914ee8a1df2a0dec9d2b9206f04",
+				PostDutyRunnerStateRoot: notDecidedSyncCommitteeSC().Root(),
+				PostDutyRunnerState:     notDecidedSyncCommitteeSC().ExpectedState,
 				OutputMessages:          []*types.SignedPartialSignatureMessage{},
 				ExpectedError:           expectedErr,
```
Comment (GalRogozinski) [inline ssv/spectest/tests/runner/duties/newduty/not_decided.go:55]: I am actually not completely sure whe we shouldn't allow parallelism here. Maybe for some reason the node got delayed with the first sync committee duty?

**Review by GalRogozinski — COMMENTED** [review]
@olegshmuelov can you please answer whatever questions you can.
No need to change anything in the code.
Also see my response to #266

File: ssv/spectest/tests/runner/duties/newduty/decided.go [hunk 3]
```diff
@@ -18,7 +18,7 @@ func PostDecided() tests.SpecTest {
 
 	// TODO: check error
 	// nolint
```
[the remark this answers is not in the document]
Comment (GalRogozinski) [reply ssv/spectest/tests/runner/duties/newduty/decided.go:20]: when deleting the comment I didn't get an error

File: ssv/spectest/tests/runner/duties/newduty/decided.go [hunk 4]
```diff
@@ -18,7 +18,7 @@ func PostDecided() tests.SpecTest {
 
 	// TODO: check error
```
[the remark this answers is not in the document]
Comment (olegshmuelov) [reply ssv/spectest/tests/runner/duties/newduty/decided.go:20]: don't see why

File: ssv/spectest/tests/runner/duties/newduty/decided.go (hunk 1 above)
Comment (GalRogozinski) [reply ssv/spectest/tests/runner/duties/newduty/decided.go:51]: moving discussion to #271

**Review by GalRogozinski — APPROVED** [review]

File: ssv/spectest/tests/runner/duties/newduty/not_decided.go [hunk 5]
```diff
@@ -48,35 +49,28 @@ func NotDecided() tests.SpecTest {
...
 				Duty:                    &testingutils.TestingAggregatorDutyNextEpoch,
-				PostDutyRunnerStateRoot: "1748cd8b8c6625951806144579b24fb1e8bb321f685dcc33f02194076afb00d7",
+				PostDutyRunnerStateRoot: notDecidedAggregatorSC().Root(),
+				PostDutyRunnerState:     notDecidedAggregatorSC().ExpectedState,
 				OutputMessages: []*types.SignedPartialSignatureMessage{
 					testingutils.PreConsensusSelectionProofNextEpochMsg(ks.Shares[1], ks.Shares[1], 1, 1), // broadcasts when starting a new duty
 				},
 				ExpectedError: expectedErr,
 			},
-			{
```
Comment (alonmuroch) [inline ssv/spectest/tests/runner/duties/newduty/not_decided.go:65]: why is this removed?

**Review by alonmuroch — CHANGES_REQUESTED** [review]

File: ssv/spectest/tests/runner/duties/newduty/not_decided.go (hunk 5 above)
Comment (GalRogozinski) [reply ssv/spectest/tests/runner/duties/newduty/not_decided.go:65]: if you scroll down to the end of the file you will see that we use versioned proposer for blinded/non-blinded:

```
	for _, v := range testingutils.SupportedBlockVersions {
		multiSpecTest.Tests = append(multiSpecTest.Tests, []*StartNewRunnerDutySpecTest{proposerV(v), proposerBlindedV(v)}...)
	}
```

**Review by alonmuroch — APPROVED** [review]

**Review by alonmuroch — APPROVED** [review]
